### PR TITLE
Added ParseTreeTest

### PR DESCRIPTION
**Pull request by sbayer55**
Signed-off-by: Steven Bayer <smbayer@amazon.com>

### Description
Adds Data Prepper Expression parsers tests that assert full expression parsing
 
### Issues Resolved
Missing test coverage
 
### Check List
- [x] New functionality includes testing.
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
- [x] Commits are signed with a real name per the DCO

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/data-prepper/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: data-prepper-expression/src/test/java/org/opensearch/dataprepper/expression/ParseTreeTest.java [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+                EQUALITY_OPERATOR_EXPRESSION
+        ).withChildrenMatching(isUnaryTree(), isOperator(EQUALITY_OPERATOR), isUnaryTree());
+
+        assertThat(expression, isExpression(equalityExpression));
+        assertThat(errorListener.isErrorFound(), is(false));
+    }
+
+    @Test
+    void testConditionalExpression() {
+        final ParserRuleContext expression = parseExpression("false and true");
```
Comment (dapowers87) [inline data-prepper-expression/src/test/java/org/opensearch/dataprepper/expression/ParseTreeTest.java:55]: Would it be beneficial to test `"falkse and true"`, looking for typos?

File: data-prepper-expression/src/test/java/org/opensearch/dataprepper/expression/ParseTreeTest.java [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+        );
+
+        assertThat(expression, isExpression(thirdConditional));
+        assertThat(errorListener.isErrorFound(), is(false));
+    }
+
+    @Test
+    void testValidJsonPointerCharacterSet() {
+        final ParserRuleContext expression = parseExpression(
+                "/ABCDEFGHIJKLMNOPQRSTUVWXYZ/ambcdefghijklmnopqrstuvwxyz/0123456789/_");
```
Comment (dapowers87) [inline data-prepper-expression/src/test/java/org/opensearch/dataprepper/expression/ParseTreeTest.java:193]: looks like you have an errant `m` after `a`

**Review by dapowers87 — CHANGES_REQUESTED** [review]
I'm interested in your feedback in my question and then the one trivial code change.

File: data-prepper-expression/src/test/java/org/opensearch/dataprepper/expression/ParseTreeTest.java (hunk 2 above)
Comment (sbayer55) [reply data-prepper-expression/src/test/java/org/opensearch/dataprepper/expression/ParseTreeTest.java:193]: Good catch!

**Review by dapowers87 — APPROVED** [review]

**Review by dinujoh — APPROVED** [review]

**Pull request merged**